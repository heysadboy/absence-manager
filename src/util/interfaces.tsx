export interface IAbsence {
    admitterId: number | null;
    admitterNote: string | null,
    confirmedAt: string | null,
    createdAt: string | null,
    crewId: number,
    endDate: string | null,
    id: number,
    memberNote: string | null,
    rejectedAt: string | null,
    startDate: string | null,
    type: string | null,
    userId: number
}

export interface IMember {
    crewId: number,
    id: number,
    image: string,
    name: string,
    userId: number
}

export interface IMemberList {
    [id: number]: IMember
}