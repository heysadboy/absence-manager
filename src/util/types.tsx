export type TPageIndex = {
    start: number;
    end: number;
    total: number
}

export type TDateRange = {
    start: string,
    end: string
}