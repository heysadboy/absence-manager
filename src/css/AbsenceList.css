#absence-list {
    padding: 110px 0px 100px 0px;
}