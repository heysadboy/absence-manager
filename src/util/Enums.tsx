export enum EAbsenceStatus {
    requested = 'Requested',
    confirmed = 'Confirmed',
    rejected = 'Rejected'
}

export enum EPageStatus {
    loading = 'Loading',
    error = 'Error',
    success = 'Success',
    empty = 'Empty'
}

export enum EAbsenceType {
    sickness = 'Sickness',
    vacation = 'Vacation',
    all = 'All'
}
