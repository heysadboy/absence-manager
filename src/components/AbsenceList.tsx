import { FC } from 'react';
import { IMemberList, IMember, IAbsence } from '../util/interfaces';
import AbsenceItem from './AbsenceItem';

interface IAbsenceListProps {
    members: IMemberList;
    absences: IAbsence[];
}

const AbsenceList: FC<IAbsenceListProps> = ({ members, absences }) => {
    const renderAbsenceList = absences.map(absence => {
        let member: IMember = members[absence.userId];
        return (<AbsenceItem key={absence.id} absence={absence} member={member} />);
    })

    return (
        <div className="ui list">{renderAbsenceList}</div>
    )
};

export default AbsenceList;