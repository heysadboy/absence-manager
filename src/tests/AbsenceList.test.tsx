import renderer from 'react-test-renderer';
import AbsenceList from '../components/AbsenceList';
import { IAbsence, IMemberList } from '../util/Interfaces';


test('AbsenceList.tsx snapshot testing', () => {

    const members: IMemberList = {
        1: {
            crewId: 1,
            id: 1,
            image: "image link",
            name: "name 1",
            userId: 1
        },
        2: {
            crewId: 1,
            id: 1,
            image: "image link",
            name: "name 2",
            userId: 1
        }
    };

    const absences: IAbsence[] = [
        {
            admitterId: 1,
            admitterNote: "admitter note",
            confirmedAt: "2020-12-12",
            createdAt: "2020-12-12",
            crewId: 1,
            endDate: "2020-12-12",
            id: 1,
            memberNote: "member note",
            rejectedAt: "2020-12-12",
            startDate: "2020-12-12",
            type: "vacation",
            userId: 1
        },
        {
            admitterId: 2,
            admitterNote: "admitter note",
            confirmedAt: "2020-12-12",
            createdAt: "2020-12-12",
            crewId: 2,
            endDate: "2020-12-12",
            id: 2,
            memberNote: "member note",
            rejectedAt: "2020-12-12",
            startDate: "2020-12-12",
            type: "vacation",
            userId: 2
        }

    ]

    const component = renderer.create(
        <AbsenceList members={members} absences={absences} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});