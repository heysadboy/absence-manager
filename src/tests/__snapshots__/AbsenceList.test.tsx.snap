// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbsenceList.tsx snapshot testing 1`] = `
<div
  className="ui relaxed divided list"
  id="absence-list"
>
  <div
    className="ui grid"
    id="absence-item"
  >
    <div
      className="three wide column"
    >
      <img
        alt="name 1"
        className="ui avatar image"
        id="member-image"
        src="image link"
      />
    </div>
    <div
      className="thirteen wide column"
    >
      <div
        className="content"
      >
        <div
          className="ui mini green label"
        >
          Confirmed
        </div>
        <span
          id="member-name"
        >
          name 1
        </span>
        <div
          className="description"
          id="member-description"
        >
          <i
            className="home icon"
          />
           
          <span
            id="absence-type"
          >
            vacation
          </span>
          <br />
          <i
            className="calendar alternate outline icon"
          />
           
          2020-12-12
           - 
          2020-12-12
          <br />
          Member Note: member note
          <br />
          Admitter Note: admitter note
        </div>
      </div>
    </div>
  </div>
  <div
    className="ui grid"
    id="absence-item"
  >
    <div
      className="three wide column"
    >
      <img
        alt="name 2"
        className="ui avatar image"
        id="member-image"
        src="image link"
      />
    </div>
    <div
      className="thirteen wide column"
    >
      <div
        className="content"
      >
        <div
          className="ui mini green label"
        >
          Confirmed
        </div>
        <span
          id="member-name"
        >
          name 2
        </span>
        <div
          className="description"
          id="member-description"
        >
          <i
            className="home icon"
          />
           
          <span
            id="absence-type"
          >
            vacation
          </span>
          <br />
          <i
            className="calendar alternate outline icon"
          />
           
          2020-12-12
           - 
          2020-12-12
          <br />
          Member Note: member note
          <br />
          Admitter Note: admitter note
        </div>
      </div>
    </div>
  </div>
</div>
`;
